[INKLB-005] storage: file holder:

kingjoram opened the pull request:
https://trello.com/c/N0j3zhON/38-b-005-file-holder

kingjoram committed:
NEW:file-system:readfile:add ReadFile.h

kingjoram committed:
NEW:file-system:readfile:add read_file.cpp

kingjoram committed:
NEW:file-system:deletefile:add delete_file.h

kingjoram committed:
NEW:file-system:savefile:add save_changes.h

kingjoram committed:
FIX:file-system:readfile:close file

kingjoram committed:
NEW:file-system:savefile:add save_changes.cpp

kingjoram committed:
FIX:file-system:savefile:close file

sonofR9 commented on back/services/storage/include/delete_file.h:
```diff
@@ -0,0 +1,12 @@
+    bool Delete(std::string filePath);
+};
+
+#endif  // _DELETE_FILE_H_
```
Обычно просят пустую строку в конце файла

sonofR9 commented on back/services/storage/include/delete_file.h:
```diff
@@ -0,0 +1,12 @@
+class DeleteFile
+{
+public:
+    bool Delete(std::string filePath);
```
Лучше const std::string&. Просто std::string норм когда перемещаешь потом в свою переменную (чтобы не писать две перегрузки), а когда просто попользоваться и забыть, то const std::string&.

sonofR9 commented on back/services/storage/include/save_changes.h:
```diff
@@ -0,0 +1,13 @@
+#define _SAVE_CHANGES_H_
+
+#include <string>
+
```
В vs ctrl+k, ctrl+d (в такой последовательности) должны вызвать clang-format на весь файл. У меня, правда, и после нажатия ; работает, хз что у тебя было не так

sonofR9 commented on back/services/storage/src/read_file.cpp:
```diff
@@ -0,0 +1,27 @@
+        return "Uncorrect file path";
+    }
+
+    while (!fileRead.eof())
+    {
+        fileRead >> buffer;
+        stringFile = stringFile + buffer;
+    }
```
\n не будет прочитано, что не очень хорошо.  Плюс конец, кажется, дважды может прочитаться, т.к. eof возвращает true только когда на него перешли.

sonofR9 commented on back/services/storage/src/save_changes.cpp:
```diff
@@ -0,0 +1,20 @@
+#include <fstream>
+#include <string>
+#include "save_changes.h"
```
Обычно делят инклуды на несколько блоков (clang-format сделает у нас. Типа такого по google style-guide для файла my.cpp:
#include "my.h"

#include <std>

#include <third-party>

#include <local-lib>

#include "headers in this project.h"

Но в целом зависит от выбранного стиля)

sonofR9 commented on back/services/storage/src/read_file.cpp:
```diff
@@ -0,0 +1,27 @@
+        stringFile = stringFile + buffer;
+    }
+
+    fileRead.close();
```
RAII: fileRead не надо закрывать, он сам в деструкторе закроется.

sonofR9 reviewed (commented):
Из важного только про fileRead, остальное так


И ещё
про классы: хорошо бы сделать интерфейсы (можно оставить всё как есть, просто добавить virtual ~ClassName()= default), чтобы в тестах можно было заменить на Mock.
про названия: обычно переменные называют существительными (например, не fileRead, а fileReader)

sonofR9 committed:
Move env in cmake to project-level variables

sonofR9 committed:
Clang format wider column limit

sonofR9 committed:
Clang format include order: std last

sonofR9 committed:
Std::tie instead of multiple ==

sonofR9 committed:
Folders backend, frontend

sonofR9 committed:
REF pragma once instead of idndef...

sonofR9 committed:
REF service is not interface => no virtual

Service is executable.
It doesn't benefits from interface (including testing)

sonofR9 committed:
REF "& " instead of " &"

sonofR9 committed:
REF empty lines at the end of files

sonofR9 committed:
FIX virtual destructor in "virtual" classes

sonofR9 committed:
Frontend and backend in tests cmake

kingjoram committed:
FIX:file-system:read/save/delete:modified files after review

sonofR9 committed:
REF add namespaces

sonofR9 committed:
REF made pure virtual functions

kingjoram committed:
INKLB-005 NEW:file-system:file holder:add file holder interface

sonofR9 committed:
No network adapter for now, namespaces inside classes

sonofR9 committed:
Typo chassis, not chassiss

sonofR9 committed:
Add using, missed forward declaration

kingjoram committed:
INKLB-005 NEW:file-system:file holder:add file holder class

sonofR9 committed:
tcp::socket is nested class, can't forward declare

sonofR9 committed:
REF listener case fix

Gallaann committed:
added cmake to auth

Gallaann committed:
added cmake for draw

Gallaann committed:
added cmake for graph

Gallaann committed:
added cmake for menu

sonofR9 committed:
REF constructor params instead of setters

sonofR9 committed:
Merge branch 'initial_structure' of github.com:cpp-park-vk-education/2023_1_The_Loopers into initial_structure

sonofR9 committed:
Shared_ptr to listener constructor

Because to construct with reference, object must be saved where it is called.
For async classes in this case it is not very convenient.

sonofR9 committed:
Session factory should return shared_ptr

Async work based on memory managmenet of shared_ptr
I think, with regular ptr Session will destruct itself immediately

Gallaann committed:
Add namespace for graph component

Gallaann committed:
REF get rid of std::pair

It's not necessary to store position of vertex, because IVertex contains position itself

Gallaann committed:
Add namespace in IEdge

Gallaann committed:
Add namespace in IVertex

Gallaann committed:
REF moved Position struct from GraphModel to IVertex

Gallaann committed:
REF GraphModel now sets positions to vertexes

Gallaann committed:
REF now auth adds as library, because it's should be component of general

Gallaann committed:
REF now draw adds as library, because it's should be component of general

Gallaann committed:
REF now graph adds as library, because it's should be component of general

Gallaann committed:
REF now menu adds as library, because it's should be component of general

Gallaann committed:
Add general to the structure

Gallaann committed:
REF added missing &

Gallaann committed:
FIX order public - protected - private

Gallaann committed:
FIX deleted Point because of redefinition

Gallaann committed:
FIX added std namespace in Point

Gallaann committed:
REF made methods const

Gallaann committed:
REF add unique pointer

Gallaann committed:
REF made getters const

Gallaann committed:
FIX added missing &

Gallaann committed:
REF added namespace for auth

Gallaann committed:
REF added namespace for draw

Gallaann committed:
REF added namespace for menu

Gallaann committed:
FIX added missing & in draw

sonofR9 committed:
REF remove spaces between methods

Also std::hash specialization is inline => ok

sonofR9 committed:
Merge branch 'initial_structure' of github.com:cpp-park-vk-education/2023_1_The_Loopers into initial_structure

sonofR9 committed:
REF add m_

sonofR9 committed:
REF rename unsigned short to uint16_t

Gallaann committed:
FIX name in cmake for general

kingjoram committed:
INKLB-005 DEL:file-system:file delete:deleted delete_file.h and delete_file.cpp

kingjoram committed:
INKLB-005 ENH:file-system:file holder/save/read:add namespace

sonofR9 committed:
FIX CI: now fails on tests if tests fails

sonofR9 committed:
Merge branch 'initial_structure' of github.com:cpp-park-vk-education/2023_1_The_Loopers into initial_structure

sonofR9 committed:
CI fail on tests

sonofR9 committed:
CI, in cmake script to run tests

sonofR9 committed:
Structure in tests

sonofR9 committed:
FIX std::hash is const noexcept

sonofR9 committed:
Static libs instead of shared

Couldn't make it work (on win32) ((

sonofR9 committed:
FIX IListener namespace in chassis

Skiller17 commented on back/services/storage/src/file_holder.cpp:
```diff
@@ -0,0 +1,28 @@
+#include <string>
+#include "file_holder.h"
```
Хэдер называется по-другому(file_holred)

kingjoram replied:
исправил

Skiller17 commented on back/services/storage/src/file_holder.cpp:
```diff
@@ -0,0 +1,28 @@
+        return false;
+    }
+
+    std::string Read(std::string filePath) override
```
Почему метод называется Read, когда это обычный геттер

Skiller17 commented on back/services/storage/src/read_file.cpp:
```diff
@@ -0,0 +1,25 @@
+
+    while (fileRead >> buffer)
+    {
+        stringFile = stringFile + buffer;
```
Мне кажется нужен /n

Skiller17 commented on back/services/storage/include/file_holred.h:
```diff
@@ -0,0 +1,28 @@
+
+
+#include <string>
+#include "delete_file.h"
```
Названия разные, кстати не забудь спуллить с initial_structure свои ветки, у тебя структура устарела

kingjoram replied:
сделал

Skiller17 commented on back/services/storage/src/file_holder.cpp:
```diff
@@ -0,0 +1,28 @@
+{
+    bool Save(std::string filePath, std::string fileChanges) override
+    {
+        if (m_changesSavior.Save(filePath, fileChanges))
```
Очень странная тема, как я понял у тебя объект m_changesSavior в классе SaveChanges имеет метод Save, но когда ты пихаешь его внутрь другого метода Save выглядит как минимум странно. Кстати не забывай, что методы в camelCase с маленькой буквы пишутся

Skiller17 reviewed (commented):
- Обрати внимание на camelCase
- Внимательно проверяй названия хэдеров
- Запуль изменения из initial_structure
- Разберись с названиями методов

kingjoram committed:
Merge branch 'initial_structure' of https://github.com/cpp-park-vk-education/2023_1_The_Loopers into dev/a.danilchenko/file_holder

# Conflicts:
#	backend/services/storage/include/file_holred.h
#	backend/services/storage/include/read_file.h
#	backend/services/storage/include/save_changes.h

kingjoram committed:
INKLB-005 REF:fileholder: review changes

sonofR9 committed:
FIX SERVICES_DIR is .../backend/...

sonofR9 committed:
FIX rm initialization of members in base chassis

sonofR9 committed:
FIX add socket libs on WIN32 for asio

Gallaann committed:
Initial project structure frontend + backend

kingjoram committed:
Merge branch 'main' of https://github.com/cpp-park-vk-education/2023_1_The_Loopers into dev/a.danilchenko/file_holder
